### ISSUE TITLE
Running from a fork requires matching, pre-existing branches

### ISSUE THREAD

**Issue by joeyparrish** [google-github-actions/release-please-action#819]
### TL;DR

Using `fork: true` only works if the fork already contains a branch with the same name as the upstream target branch that triggered the release action.

Now that I've run into this, I'm unsure if I'm even using it correctly.  Will release-please try to build the change list from the fork instead of the upstream?  My intent is only to open the PR from the fork, not do everything based on the fork's contents.

If I'm misusing `fork: true`, please help me understand that and please help me update the documentation to clarify its intent.

If `fork: true` is simply missing a feature to create missing branches on the fork, then please help me understand whether I should attempt to add that feature.

### Expected behavior

I would expect release-please-action to create that branch if needed.

### Observed behavior

release-please-action fails with `Branch not found`.

### Action YAML

```yaml
name: Release

on:
  push:
    branches:
      - main
      - v[0-9]*

jobs:
  release:
    runs-on: ubuntu-latest
    outputs:
      release_created: ${{ steps.release.outputs.release_created }}
      tag_name: ${{ steps.release.outputs.tag_name }}
    steps:
      # Create/update release PR
      - uses: google-github-actions/release-please-action@v3
        id: release
        with:
          # Required input to specify the release type (node package).
          release-type: node
          # Make sure we create the PR against the correct branch.
          default-branch: ${{ github.ref_name }}
          # Use a special shaka-bot access token for releases.
          token: ${{ secrets.RELEASE_PLEASE_TOKEN }}
          # Create the PR from the bot account's fork.
          fork: true

      # If we didn't create a release, we may have created or updated a PR.
      # Check out the code, then update the Player version in the PR.
      - uses: actions/checkout@v3
        if: steps.release.outputs.release_created == false
        with:
          # Checkout the fork, where the PR is generated from.
          repository: shaka-bot/${{ github.event.repository.name }}
          # Use a special shaka-bot access token for releases.
          token: ${{ secrets.RELEASE_PLEASE_TOKEN }}
      - name: Custom update Player version
        if: steps.release.outputs.release_created == false
        run: |
          # Check out the branch that release-please created.
          # If it does not exist, FAIL!
          git fetch
          git checkout release-please--branches--${{ github.ref_name }}--components--shaka-player || exit 1
          # If it does exist, update lib/player.js in the PR branch, so that the
          # -uncompiled tag remains in the player version in that context.
          VERSION="v$(jq -r .version package.json)-uncompiled"
          sed -e "s/^\\(shaka.Player.version =\\).*/\\1 '$VERSION';/" \
              -i lib/player.js
          git add lib/player.js
          # Set missing git config for the commit.
          git config user.name "shaka-bot"
          git config user.email "shaka-bot@users.noreply.github.com"
          # Update the PR.
          git commit --amend --no-edit
          git push -f

  # The jobs below are all conditional on a release having been created by
  # someone merging the release PR.  They all run in parallel.

  tag-main:
    runs-on: ubuntu-latest
    needs: release
    if: needs.release.outputs.release_created && endsWith(needs.release.outputs.tag_name, '.0') == false
    steps:
      - uses: actions/checkout@v3
        with:
          # Check out the origin repo, not the fork, and do it at main, not the
          # PR branch.
          ref: main
          # Use a special shaka-bot access token for releases.
          token: ${{ secrets.RELEASE_PLEASE_TOKEN }}
      - name: Tag the main branch
        run: |
          # Set missing git config for the tag.
          git config user.name "shaka-bot"
          git config user.email "shaka-bot@users.noreply.github.com"
          # Tag the main branch.
          VERSION=${{ needs.release.outputs.tag_name }}
          git tag -m "$VERSION-main" "$VERSION-main"
          git push origin "$VERSION-main"

  npm:
    runs-on: ubuntu-latest
    needs: release
    if: needs.release.outputs.release_created
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          persist-credentials: false

      - uses: actions/setup-node@v1
        with:
          node-version: 16
          registry-url: 'https://registry.npmjs.org'

      - run: npm ci
      - run: npm publish
        env:
          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}

      - run: npm pack
      - uses: svenstaro/upload-release-action@483c1e56f95e88835747b1c7c60581215016cbf2
        with:
          repo_token: ${{ secrets.GITHUB_TOKEN }}
          tag: ${{ needs.release.outputs.tag_name }}
          file: shaka-player-*.tgz
          file_glob: true
          overwrite: true

  appspot:
    runs-on: ubuntu-latest
    needs: release
    if: needs.release.outputs.release_created
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          persist-credentials: false

      - uses: actions/setup-node@v1
        with:
          node-version: 16
          registry-url: 'https://registry.npmjs.org'

      - name: Compute appspot subdomain and promotion
        run: |
          # This is the same as the version tag, but with dots replaced by
          # dashes.  For example, v3.2.2 would have the subdomain v3-2-2.
          APPSPOT_SUBDOMAIN=$( echo ${{ needs.release.outputs.tag_name }} | sed -e 's/\./-/g' )
          echo APPSPOT_SUBDOMAIN=$APPSPOT_SUBDOMAIN >> $GITHUB_ENV

          # "Promoting" an appspot deployment makes it the default which shows
          # up on shaka-player-demo.appspot.com (no subdomain).  This should be
          # done for the latest release version from the latest release branch.
          RELEASE_TAGS=$(git tag | grep ^v[0-9] | grep -Ev -- '-(master|main)')
          LATEST_RELEASE=$(echo "$RELEASE_TAGS" | sort --version-sort | tail -1)
          TAG_NAME=${{ needs.release.outputs.tag_name }}

          if [[ "$TAG_NAME" == "$LATEST_RELEASE" ]]; then
            echo APPSPOT_PROMOTE=true >> $GITHUB_ENV
          else
            echo APPSPOT_PROMOTE=false >> $GITHUB_ENV
          fi

          # Debug the decisions made here.
          echo "Subdomain: $APPSPOT_SUBDOMAIN"
          echo "Latest release: $LATEST_RELEASE"
          echo "This release: $TAG_NAME"
          echo "Promote: $APPSPOT_PROMOTE"

      - uses: ./.github/workflows/custom-actions/prep-for-appspot

      - uses: google-github-actions/auth@v0
        with:
          credentials_json: '${{ secrets.APPENGINE_DEPLOY_KEY }}'

      - uses: google-github-actions/deploy-appengine@v0
        with:
          project_id: shaka-player-demo
          version: ${{ env.APPSPOT_SUBDOMAIN }}
          promote: ${{ env.APPSPOT_PROMOTE }}

  auto-branch:
    runs-on: ubuntu-latest
    needs: release
    if: needs.release.outputs.release_created && endsWith(needs.release.outputs.tag_name, '.0')
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          # Use a special shaka-bot access token for releases.
          token: ${{ secrets.RELEASE_PLEASE_TOKEN }}

      - name: Create release branch
        run: |
          TAG=${{ needs.release.outputs.tag_name }}
          BRANCH=$(echo "$TAG" | sed -e 's/\.0$/.x/')
          git push origin HEAD:"$BRANCH"
```

### Log output

```text
With the full logs included inline, I get `There was an error creating your issue: body is too long (maximum is 65536 characters)` from GitHub when I try to submit the issue.  So I have attached them in the "additional information section".

Logs from running without a pre-existing branch to match the upstream target (https://github.com/shaka-project/shaka-player/actions/runs/6055899919/job/16435703152) in logs_42143.zip.

Logs from re-running that job after manually creating a matching branch (https://github.com/shaka-project/shaka-player/actions/runs/6055899919/job/16435752138) in logs_42144.zip.
```

### Additional information

[logs_42143.zip](https://github.com/google-github-actions/release-please-action/files/12503095/logs_42143.zip)

[logs_42144.zip](https://github.com/google-github-actions/release-please-action/files/12503094/logs_42144.zip)

**Comment by chingor13** [comment]
Currently release-please (really https://github.com/googleapis/code-suggester) does not keep the fork up to date once the initial forking is done. I believe there is an API to do so, but I'm not sold that this should be on every release PR created (versus a one-off sync).

The change would need to be in the upstream release-please library (https://github.com/googleapis/release-please) or in code-suggester which is the library doing the forking and opening the PR.
